apps/src/flowlab/views/flow/nodes/AiNode.tsx

```diff
@@ -0,0 +1,124 @@
+import FontAwesomeV6Icon from '@code-dot-org/component-library/fontAwesomeV6Icon';
+import {
+  Position,
+  Handle,
+  useReactFlow,
+  useNodeConnections,
+  useNodesData,
+  type Node,
+  type NodeProps,
+} from '@xyflow/react';
+import React, {memo, useEffect, useState, useCallback} from 'react';
+
+import askAi from '../../../flow/askAi';
+import {isTextNode, type MyNode} from '../../../flow/initialElements';
+
+// This node asks the aichat service a question entered in its input field, and
+// sends the response to its output handle as text.  It acepts optional context
+// via any text sent to its input handle.  It makes a new request whenever
+// shift-enter is pressed while its input has focus, and whenever its input
+// context changes.
+
+function AiNode({
+  id,
+  data,
+}: NodeProps<Node<{fieldText: string; askedText: string; text: string}>>) {
+  const {updateNodeData} = useReactFlow();
+  const connections = useNodeConnections({
+    handleType: 'target',
+  });
+  const nodesData = useNodesData<MyNode>(
+    connections.map(connection => connection.source)
+  );
+  const textNodes = nodesData.filter(isTextNode);
+
+  const [isWorking, setIsWorking] = useState(false);
+
+  const [lastContextString, setLastContextString] = useState<
+    string | undefined
+  >(undefined);
+
+  const contextString =
+    textNodes.length > 0
+      ? textNodes
+          .map(({data}) => (data && 'text' in data ? data.text : ''))
+          .join('')
+      : '';
```

Comment:
For my understanding is this summing all the text inputs into the current AI node?